pageserver/src/tenant.rs

```diff
@@ -452,19 +485,6 @@ impl Tenant {
             }
         };
 
-        if self.remote_storage.is_some() {
-            // Reconcile local state with remote storage, downloading anything that's
-            // missing locally, and scheduling uploads for anything that's missing
-            // in remote storage.
-            timeline
-                .reconcile_with_remote(
-                    up_to_date_metadata,
-                    remote_startup_data.as_ref().map(|r| &r.index_part),
-                )
-                .await
-                .context("failed to reconcile with remote")?
-        }
-
         // Sanity check: a timeline should have some content.
         anyhow::ensure!(
             ancestor.is_some()
```

Comment:
This I needed to check, looking now.

Reply:
> 1. I'm not sure that further down code is needed anymore

It's about reflecting the remote metadata locally. While on the next restart we could come to same conculusion, it feels like it ought to be done.

> 2. If it is, it's not scheduling an index file upload

If we picked remote, is there some need?

> 3. Shouldn't we always persist local metadata before scheduling upload?

Looking at the code, this seems quite spread around.

- gc: explicit Timeline::update_metadata_file first, later file deletions and scheduling deletes
    - "later" grows unbound with #4938
    - needed because `latest_gc_cutoff` grows
- compact_level0: cannot see any explicit metadata update
    - remote update is handled via uploads and (later) deletions
    - explicit index upload on success and later with deletions
    - last_record_lsn or disk_consistent_lsn does not advance so this is probably all good but with re-write hazard
    - could track compaction progress LSN in the metadata to avoid rewrite hazard
- flush_frozen_layer: explicit Timeline::update_metadata_file and upload(s)

Reply:
5e931fec0 moves the `if !picked_local` to happen before loading layer map and init remote timeline client.

Reply:
That feels less weird. I guess what makes it feel weird is that 

* during attach, we know we're always saving the remote metadata
* during load, we know we're never saving the remote metadata

If only I had completed my https://github.com/neondatabase/neon/pull/3466